packages/symantec_endpoint_security/data_stream/event/elasticsearch/ingest_pipeline/default.yml

```
@@ -1105,6 +1105,25 @@ processors:
             - append:
                 field: error.message
                 value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+  - script:
+      description: Move raw_data to raw_data.value if the value is not compatible with flattened.
+      lang: painless
+      if: ctx.ses?.raw_data != null && !(ctx.ses.raw_data instanceof Map)
+      source: |-
+        if (ctx.ses.raw_data instanceof List) {
+          for (def e: ctx.ses.raw_data) {
+            if (!(e instanceof Map)) {
+              def m = new HashMap();
+              m["value"] = ctx.ses.raw_data;
```

```suggestion
              m["value"] = e;
```
Not sure if its meant to be this way?

---

No, what's there is correct. We're looking for anything that is not a map and if we find it we know that the whole thing must be put into a value field to that it is a map. I think the alternative that you have is also valid, but I chose the way that I did to minimize changing that value's shape. I don't think it will make much difference in reality since this was purely to cover cases that might happen, but which we have not seen.

---

thanks for the clarification